oct_converter/dicom/dicom.py

```diff
@@ -0,0 +1,292 @@
+        if meta.series_info.acquisition_date
+        else ""
+    )
+    ds.AcquisitionNumber = 1
+    ds.PhotometricInterpretation = "MONOCHROME2"
+    # Unsigned integer
+    ds.PixelRepresentation = 0
+    # Use 16 bit pixel
+    ds.BitsAllocated = 16
+    ds.BitsStored = ds.BitsAllocated
+    ds.HighBit = ds.BitsAllocated - 1
+    ds.SamplesPerPixel = 1
+    ds.NumberOfFrames = len(frames)
+
+    # Multi-frame Functional Groups Module PS3.3 C.7.6.16
+    dt = datetime.now()
+    ds.ContentDate = dt.strftime("%Y%m%d")
+    timeStr = dt.strftime("%H%M%S.%f")  # long format with micro seconds
+    ds.ContentTime = timeStr
+    ds.InstanceNumber = 1
+
+    per_frame = []
+    pixel_data_bytes = list()
+    # Convert to a 3d volume
+    pixel_data = np.array(frames).astype(np.uint16)
+    ds.Rows = pixel_data.shape[1]
+    ds.Columns = pixel_data.shape[2]
+    for i in range(pixel_data.shape[0]):
+        # Per Frame Functional Groups
+        frame_fgs = Dataset()
+        frame_fgs.PlanePositionSequence = [Dataset()]
+        ipp = [0, 0, i * meta.image_geometry.slice_thickness]
+        frame_fgs.PlanePositionSequence[0].ImagePositionPatient = ipp
+        frame_fgs.FrameContentSequence = [Dataset()]
+        frame_fgs.FrameContentSequence[0].InStackPositionNumber = i + 1
+        frame_fgs.FrameContentSequence[0].StackID = "1"
+
+        # Pixel data
+        frame_dat = pixel_data[i, :, :]
+        pixel_data_bytes.append(frame_dat.tobytes())
+        per_frame.append(frame_fgs)
+    ds.PerFrameFunctionalGroupsSequence = per_frame
+    ds.PixelData = pixel_data.tobytes()
+    ds.save_as(file_)
+    return file_
+
+
+def create_dicom_from_oct(
+    input_file: str, output_dir: str = None, output_filename: str = None
+) -> Path:
+    """Creates a DICOM file with the data parsed from
+    the input file.
+
+    Args:
+            input_file: File with OCT data (Currently only Topcon
+            files supported)
+            output_dir: Output directory, will be created if
+            not currently exists. Default None places file in
+            current working directory.
+            output_filename: Name to save the file under, i.e.
+            `filename.dcm`. Default None saves the file under
+            the input filename (if input_file = `test.fds`,
+            output_filename = `test.dcm`)
+
+    Returns:
+            Path to DICOM file
+    """
+    file_suffix = input_file.split(".")[-1]
+    if file_suffix == "fds":
+        fds = FDS(input_file)
+        oct = fds.read_oct_volume()
+        meta = fds_dicom_metadata(oct)
+    elif file_suffix == "fda":
+        fda = FDA(input_file)
+        oct = fda.read_oct_volume()
+        meta = fda_dicom_metadata(oct)
+    elif file_suffix in ["e2e", "img", "oct", "OCT"]:
+        raise NotImplementedError("Filetype not yet supported.")
+    else:
+        raise TypeError("Invalid input_file type.")
+    file = write_opt_dicom(meta, oct.volume)
```

Comment:
I'll pull @naterichman in on this; he originally put together the base of the dicom creation code, and would have better insights than me!

Reply:
Updated to 1) no longer save to a temporary location and move, and 2) only use pathlib instead of os for consistency. Let me know if anything else about this part sticks out as needing improvement!

Reply:
i think we need        
 output_dir.mkdir(parents=True, **exists_ok=True)**
or running the code twice with `output_dir` set to the same value will throw an error. apart from that looks great!